tm2/pkg/std/coin.go

```diff
@@ -616,8 +616,7 @@ func (coins Coins) Sort() Coins {
 // Parsing
 
 var (
-	// Denominations can be 3 ~ 16 characters long.
-	reDnmString = `[a-z][a-z0-9]{2,15}`
+	reDnmString = `[a-z][a-z0-9/:._-]{2,127}` // cosmos regex(https://github.com/cosmos/cosmos-sdk/blob/b2c26cdc4c3d2f5b7cb28e5153e2b458c61ee8fa/types/coin.go#L852) without capital letters
```

Comment:
cc/ @jaekwon on this regex.

Reply:
> This effectively limits the realm path length, as if it's longer than 128 then it will not be able to issue coins.
> 
> Putting a limit to the realm path sounds sensible, though, so we should tackle it in another PR. For now, it probably makes sense to remove the check here, since we check already for denom against the regex in `banker.go`? (see prev comment.)

PR #1813 is checking whether string is valid or not or pkg_path, length limit can be done in there also

Reply:
BTW, I'll remove checking mechanism here

Reply:
e001ac78b4a8bc898f339fde191f56a0bbc7e158

Actually, I couldn't remove verification entirely. There was bit of testing that relies on it.